src/config.rs

```diff
@@ -29,6 +29,8 @@ pub(crate) struct Config {
   pub(crate) subcommand:           Subcommand,
   pub(crate) unsorted:             bool,
   pub(crate) unstable:             bool,
+  pub(crate) dotenv_filename:      String,
+  pub(crate) dotenv_path:          PathBuf,
```

Comment:
I think both of these will need to be options, and should be `None` if the flags aren't passed:

```suggestion
  pub(crate) dotenv_filename:      Option<OsString>,
  pub(crate) dotenv_path:          Option<PathBuf>,
```

We'll need to check if the flags were passed explicitly to determine if we should emit the deprecation warning.